analyzers/tests/SonarAnalyzer.UnitTest/Rules/InfiniteRecursionTest.cs

```diff
@@ -29,9 +30,21 @@ namespace SonarAnalyzer.UnitTest.Rules
     public class InfiniteRecursionTest
     {
         [TestMethod]
-        public void InfiniteRecursion() =>
+        public void InfiniteRecursion_SonarCfg() =>
             Verifier.VerifyAnalyzer(
-                @"TestCases\InfiniteRecursion.cs",
+                @"TestCases\InfiniteRecursion.SonarCfg.cs",
+                new InfiniteRecursion(AnalyzerConfiguration.AlwaysEnabledWithSonarCfg),
+#if NETFRAMEWORK
+                ParseOptionsHelper.OnlyCSharp7,
```

Comment:
Why not `BeforeCSharp8` instead? We're trying to test all language versions because we can have surprises. It already happened that one token had one SyntaxKind in one version and another in newer version.

Reply:
Because it will fail, since a number of the cases will fail the compilation, among them there are issues to be raised.